src/PortingAssistantExtensionServer/Services/PortingService.cs

```diff
@@ -59,22 +59,20 @@ public ProjectFilePortingResponse PortingProjects(ProjectFilePortingRequest requ
         {
             try
             {
+                if (ProjectPathToDetails == null || ProjectPathToDetails.Count == 0)
+                {
+                    return new ProjectFilePortingResponse()
+                    {
+                        Success = false,
+                        messages = new List<string>() { "please run a full assessment before porting" },
+                        SolutionPath = request.SolutionPath
+                    };
+                }
 
                 var portingRequst = new PortingRequest
                 {
-                    Projects = request.ProjectPaths.Select(p =>
-                        ProjectPathToDetails.TryGetValue(p, out var projectDetails) ? projectDetails
-                        : new ProjectDetails()
-                        {
-                            ProjectName = "",
-                            ProjectFilePath = p,
-                            ProjectGuid = "",
-                            ProjectType = "",
-                            TargetFrameworks = new List<string>(),
-                            PackageReferences = new List<PackageVersionPair>(),
-                            ProjectReferences = new List<ProjectReference>(),
-                            IsBuildFailed = false
-                        }).ToList(),
+
+                    Projects = ProjectPathToDetails.Select(p => p.Value).ToList(),
```

Comment:
minor comment ProjectPathToDetails.Select(p => p.Value)?.ToList(), if there are no value, ToList() will throw null ref exception

Reply:
the contents are filled from list previously... it shouldn't have null value but I'll add a check for safe. thanks